src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -2315,7 +2315,8 @@ private static IEnumerable<Diagnostic> FilterDiagnosticsByLocation(IEnumerable<D
             if (stage == CompilationStage.Declare || (stage > CompilationStage.Declare && includeEarlierStages))
             {
                 var declarationDiagnostics = GetSourceDeclarationDiagnostics(syntaxTree, filterSpanWithinTree, FilterDiagnosticsByLocation, cancellationToken);
-                Debug.Assert(declarationDiagnostics.All(d => d.HasIntersectingLocation(syntaxTree, filterSpanWithinTree)));
+                // re-enable the assert once https://github.com/dotnet/roslyn/issues/21185 is fixed
+                // Debug.Assert(declarationDiagnostics.All(d => d.HasIntersectingLocation(syntaxTree, filterSpanWithinTree)));
```

Comment:
this assert is important.  We are definitely breaking something (or have discovered something broken) if we're doing something that is causing this assert to fire.